packages/calypso-e2e/src/lib/pages/site-editor-page.ts

```diff
@@ -0,0 +1,170 @@
+import { Locator, Page } from 'playwright';
+import {
+	BlockInserter,
+	EditorGutenbergComponent,
+	EditorSidebarBlockInserterComponent,
+	EditorToolbarComponent,
+	EditorWelcomeTourComponent,
+} from '..';
+import { getCalypsoURL } from '../../data-helper';
+import envVariables from '../../env-variables';
+
+const wpAdminPath = 'wp-admin/themes.php';
+
+const selectors = {
+	editorIframe: `iframe.is-loaded[src*="${ wpAdminPath }"]`,
+	editorRoot: 'body.block-editor-page',
+	editorCanvasIframe: 'iframe[name="editor-canvas"]',
+	editorCanvasRoot: '.wp-site-blocks',
+	templateLoadingSpinner: '[aria-label="Block: Template Part"] .components-spinner',
+};
+
+/**
+ * Represents an instance of the FSE site editor.
+ * This class is composed of editor components, combining them into larger flows.
+ */
+export class SiteEditorPage {
+	private page: Page;
+
+	private editorToolbarComponent: EditorToolbarComponent;
+	private editorGutenbergComponent: EditorGutenbergComponent;
+	private editorSidebarBlockInserterComponent: EditorSidebarBlockInserterComponent;
+	private editorWelcomeTourComponent: EditorWelcomeTourComponent;
+
+	/**
+	 * Constructs an instance of the page POM class.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+
+		this.editorToolbarComponent = new EditorToolbarComponent( page, this.editor );
+		this.editorWelcomeTourComponent = new EditorWelcomeTourComponent( page, this.editor );
+		this.editorSidebarBlockInserterComponent = new EditorSidebarBlockInserterComponent(
+			page,
+			this.editor
+		);
+		// Because of the unique extra iframe in the site editor, this component needs the canvas locator.
+		this.editorGutenbergComponent = new EditorGutenbergComponent( page, this.editorCanvas );
```

Comment:
I might be missing something, but where is `this.editorCanvas` set?

Reply:
[Here](https://github.com/Automattic/wp-calypso/pull/62471/files#diff-3add2a09fe4bd7af3cca6f83125696c77757e9ca1137628c23af9c517160658dR65)!

Reply:
> [Here](https://github.com/Automattic/wp-calypso/pull/62471/files#diff-3add2a09fe4bd7af3cca6f83125696c77757e9ca1137628c23af9c517160658dR65)!

The link went to the top level PR for me :(

Reply:
Ack, sorry!

[Does this one work](https://github.com/Automattic/wp-calypso/blob/pw/editor-tracking-first-fse-tests/packages/calypso-e2e/src/lib/pages/full-site-editor-page.ts#L65)?

Reply:
Apparently it doesn't, but that's ok!